### PR TITLE
burn damge projectiles now do burn damage to cyborgs

### PR DESCRIPTION
**Pull request by imsxz**
:cl:imsxz
balance: cyborgs now take burn damage from burn damage projectiles
/:cl:
balance tag because this shit was probably intentional but damn it makes no sense, borgs shouldnt get the luxury of self healing the damage they got from that emitter beam in SM room.

### CODE REVIEW

**Comment by FrostyFridge** [comment]
yeah it makes sense I suppose.

**Comment by zxaber** [comment]
It makes sense from a lore/whatever viewpoint, but balance is going to be heavily impacted when borgs can no longer self repair from the most common weapon type on the station.

File: code/modules/mob/living/silicon/silicon_defense.dm [hunk 1]
```diff
@@ -112,13 +112,20 @@
...
+	if(Proj.damage_type == BRUTE)
 		adjustBruteLoss(Proj.damage)
 		if(prob(Proj.damage*1.5))
 			for(var/mob/living/M in buckled_mobs)
 				M.visible_message("<span class='boldwarning'>[M] is knocked off of [src]!</span>")
 				unbuckle_mob(M)
 				M.Knockdown(40)
+	if(Proj.damage_type == BURN)
+		adjustBurnLoss(Proj.damage)
+		if(prob(Proj.damage*1.5))
```
Comment (ChangelingRain) [inline code/modules/mob/living/silicon/silicon_defense.dm]: Find a way to do this without copying this section of code.

**Comment by zxaber** [comment]
I suppose I should specify; when I refer to weapons, I specifically meant firearms.

The game is _designed_ that borgs are occasionally at odds with security. That's fine, it makes for fun interactions. But security will likely be carrying lasers over ballistics. Currently borgs can tank a few shots without too much trouble, allowing them to physically block human harm and weld themselves up later. So it is worth mentioning that this will have large balance changes; it's not as simple as a human just using burn patches over brute.

**Comment by GunHog** [comment]
Also note that burn damage is significantly more effort to fix as well. Brute damage is fixed with a welding tool, while burn requires opening up the borg all the way to the wires stage (unlock via ID or self command -> crowbar cover -> remove cell [other borgs cannot do this step.] ->screwdriver to expose wires -> add cable coil to repair wires).

**Comment by zombiegums** [comment]
I'm biased because I'm a silicon main, so take what I say with a pinch of salt. I primarily dislike this PR because laser weapons are so plentiful on the station. Given that their ammo is infinite in the sense that it can be replenished with rechargers, it makes them a major and almost constant threat to borgs (which is fine). As you're already aware, given that you play borg - self-healing isn't instantaneous. It takes several seconds of standing still in one spot. If you've damaged a borg to the point where they feel like they need to do this, you've made them vulnerable and pretty much toothless for the moment. 

If I do experience burn damage, it's usually a toss-up as to whether someone will be kind enough to repair me. I have to hope for the following: robotics or other players care enough about fixing borgs to replace my wires, robo has popped a self-repair module in me so that it'll eventually heal the burn damage, or someone's upgraded the borg chargers so that they can repair all borg damage.  I've gleaned that a sizable portion of players find repairing borg burn damage off-putting because it isn't as straightforward as using a welder. It requires multiple steps and also tools (crowbar, screwdriver, cable coil) that not everyone will have on-hand/be able to afford.

I'm unsure why the emitter example was used here, as in my several hundred hours of playing borg I've never see one used offensively against a borg, nor do I think anyone cares if a borg accidentally walks in front of one and quickly fixes that damage. It feels like this is more to do with laser weapons used for combat. Gonna be honest here and say that this feels like a roundabout way of pushing the "remove the borg self-heal" PR that was rejected without outright removing the self-heal. Thematically, the change makes sense but from a gameplay perspective it makes things very unfun.

File: code/modules/mob/living/silicon/silicon_defense.dm [hunk 2]
```diff
@@ -112,13 +112,15 @@
...
-	if((Proj.damage_type == BRUTE || Proj.damage_type == BURN))
+	if(Proj.damage_type == BRUTE)
 		adjustBruteLoss(Proj.damage)
-		if(prob(Proj.damage*1.5))
-			for(var/mob/living/M in buckled_mobs)
-				M.visible_message("<span class='boldwarning'>[M] is knocked off of [src]!</span>")
-				unbuckle_mob(M)
-				M.Knockdown(40)
+	if(Proj.damage_type == BURN)
+		adjustFireLoss(Proj.damage)
```
Comment (Iamgoofball) [inline code/modules/mob/living/silicon/silicon_defense.dm]: Why do they even snowflake this crap? This should probably just get removed entirely in favor of borgs taking damage from the bullet itself, let the bullet handle it.

**Comment by imsxz** [comment]
emitter was an example for engiborgs accidentally getting hit by them during engine setup. how often are you shot by lasers as a borg? i have ~100h as one and I've only been shot at by them, well... not much, in memory. the biggest killer to me, be it from revolutionaries, operatives, or angry officers, was the stun+damage strat.

Really think about it. Lasers are the most common ranged lethal available to normal crew, yes, but how often are you facing that? Most of the time, antagonists will be packed with ballistics or EMP's, cyborgs are typically only targetted by antagonists save borgs being rogue.

The autonomy of cyborgs is my primary issue with them, they don't need to be tended to by humans after they're created, unless they end up fully dead.

File: code/modules/mob/living/silicon/silicon_defense.dm (hunk 2 above)
Comment (imsxz) [reply code/modules/mob/living/silicon/silicon_defense.dm]: someone else can refactor it i aint a coder

**Comment by Barhandar** [comment]
>emitter was an example for engiborgs accidentally getting hit by them during engine setup

And you want to fuck borgs over with accidental damage human would snack on a single warm donkpocket to fix.
>The autonomy of cyborgs is my primary issue with them, they don't need to be tended to by humans after they're created

That's the _point_ of cyborgs.

**Comment by Arianya** [comment]
This feels like its less a PR aiming at realism of burn weapons causing burn damage and more a backdoor method of cutting down silicon means to self-repair.

I don't think this PR would be acceptable without reducing the burden of repairing burn damage (which has its own problems, since EMP damage is burn damage and is intended to be hard to fix) or giving a means for borgs to do it to each other (but not self-repair) (also has EMP issues, but less so then the first option)

**Comment by vuonojenmustaturska** [comment]
As someone who plays borg fairly often, I don't see any huge cause for alarm with this PR. I guess you might not get away with fighting against blobs or mechas that other people are shooting. Security doesn't use lasers against borgs though, they use the ion rifle, flashbangs and flashes.

The flipside is that borg burn damage is roughly equivalent to carbon clone damage on a station without a medbay.

**Comment by NoxVS** [comment]
The problem with not wanting borgs to be as autonomous is that there are several situations (Malf AI, rogue AI, emagged) where borgs are forced to maintain themselves without humans. With malf AIs a lucky laser is permanent damage.

**Comment by Repukan** [comment]
Good change, for how powerful borgs are, they should be limited in their ability to self sustain

**Comment by zxaber** [comment]
vuonojenmustaturska, the steps to repair a borg's burn damage are roughly equivalent to most surgeries. Imagine having to get your appendix removed to heal burn damage every time you get shot from a laser. 

If lasers did nigh-unrepairable damage (which is what this PR is asking for in a malf/rogue situation), then Security would very much be using them against borgs. I daresay the ion rifle would never be used again.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think you meant to say "surgery but every step is instant", because the difference between the two in terms of time is quite a leap.

Only thing I dislike about this is that borgs literally cannot heal from this w/o human assistance.

**Comment by Tlaltecuhtli** [comment]
cyborg chargers when upgraded also repair borgs

**Comment by ExcessiveUseOfCobblestone** [comment]
*screwdrive crowbars the only rechargers you upgraded*

heh, nothing personal

**Comment by zombiegums** [comment]
I face getting hit by lasers when I'm not an antag quite often. About 2 or 3 days ago, I got shot in the hallway leading to brig by a stray laser shot because an officer was blind firing lethal shots down the halls for some reason? It was a mild annoyance, but I would've been more irritated if that damage had been burn over brute, given that burn is more tedious to fix.

Borgs frequently get shot at for following their laws. If harm is going down at the brig and the AI is apprised of it, we all know that it will be shouting over binary and telling its borgs to stop the harm ASAP. Good sec officers will flash a borg and drag them away and will only resort to doing damage if the borg is persistent. Others will flash and glock 'em immediately because they're an annoyance. Laser pistols are usually sec's go-to weapon. People also tend to have spaghetti aim. There have been many times during fights like blob where I'm going around dragging critted crew away and trying to stabilise them where someone has just kept shooting even though it's clearly hitting and hurting me. I'm also kinda confused at the comment stating that officers do not use lasers against borgs. They very much do, given that lasers have range, unlike borgs who generally have to attack in melee range. 

The whole point of borgs is that they're meant to be autonomous. They're machines. Sentient ones, yes - but machines. If this goes through, whatever. Mostly posting to share my POV after over 700+ hours of borg playtime because the notion that lasers aren't used against borgs is patently untrue. They're even mentioned as a threat on the malf AI wiki page. I just feel that yeah, engineering borgs and to a lesser degree, standard and mining borgs can be bullshit to fight against (since those are the only borgs that  can self-heal without an upgrade), but this majorly screws over borgs that do not have the ability to heal themselves without outside help. My main module is medical borg. Sadly, I get hurt very often and am reliant on others to get me fixed. With burn damage, there are times when I have to contend with not getting that damage fixed, even more so now that I can no longer vend the items needed to fix such such damage (crowbar, screwdriver and cable coil) myself.

**Comment by SouDescolado** [comment]
Honestly, this feels like an "I ded" issue, but maybe this should be reviewed.

For reference, this was defined as so in https://github.com/tgstation/tgstation/pull/20461, so maybe things got added after that and it broke balance. However I don't feel that borgs are overpowered. I feel this is an unnecessary change that will cause more harm(to borgs and borg sympathizer) than good(Antags have more tools to disable borgs(?))

**Comment by imsxz** [comment]
@barhandar is that the point of cyborgs? I certainly wasn’t around when they were added, I wouldn’t know. When I see cyborgs, I can’t imagine they were designed to be entire replacements for humans but with free all access.

I can only imagine they exist to fill in the gaps that humanity can’t close, vital gaps like opening tech storage so you can get a chem dispenser for hydroponics, or sustaining/curing the crew amidst a deadly disease outbreak that borgs can’t contract, enough help to get humans on their feet and moving again.

With all the assistance that silicons offer humans, there’s some maintaining to be done. Half(?) of the borg modules can’t get repairs without bugging SOMETHING else, be it a borg, human, free drone, et cetera. Hell, silicons can’t even be revived OR built without human intervention, not mentioning upgrades or cosmetics borgs may desire. There’s also their whole issue about borgs having no hands. Borgs need humans, they’re certainly NOT designed to be autonomous.

I almost forgot what I was originally arguing about here but that’s my view on cyborgs and their autonomy(or lack there of), though if you believe they are meant to be autonomous I’d be more than pleased to read your reasoning.

I wouldn’t be against making burn damage easier to repair on borgs, and would do it myself if a maintainer doesn’t find it outside of the scope of this PR.

**Comment by imsxz** [comment]
To clarify, I wouldn’t have issues with making burn damage just require slapping borgs with a cable.

I realize it appears like an i ded/imsxz is silicon hitler PR but it was mainly my autism flaring up at the dumb inconsistency, you’ll often see people trying to weld borgs or hit them with cable to fix their burn damage, this is because stuff like prosthetics only requires slapping with cable.

That said, if external burn damage was made to be easily fixable, a replacement internal damage for stuff like EMP would be of interest I figure.

I think that’s a logical compromise, please leave your thoughts on it

**Comment by Iamgoofball** [comment]
why are people getting upset that we're making cyborgs actually respect damage types properly smh

**Comment by XDTM** [comment]
Probably because burn damage is far more complicated to fix.
A potential idea is to make burn damage healable in a simpler (but different) way, and making EMP damage deal "clone" damage that is fixed by replacing wires, like current burn damage. After all external burns are not the same thing as burnt internal circuitry.

**Comment by zxaber** [comment]
>I realize it appears like an i ded/imsxz is silicon hitler PR but it was mainly my autism flaring up at the dumb inconsistency, you’ll often see people trying to weld borgs or hit them with cable to fix their burn damage, this is because stuff like prosthetics only requires slapping with cable.

That's fair, but we should be balancing around gameplay before realism. Truth be told, since borgs can walk through plasma fires without issue, I would argue the real consistency fix would be making borgs immune to lasers altogether. That aside, heat damage from the outside would likely cause warping or melted holes in the borg's shell, which would fall into the territory of welding, not cable replacement. (And the wire damage fluff referencing a warped shell ought to be re-worded, but that's neither here nor there.)

The point of borgs certainly can't be to make players act like pets that need something every other minute. They deserve a bit of self-reliance, even if it only extends to the silicon group as a whole. If Security is executing humans, and the silicons have to respond, it is helpful that they can retreat and repair if Sec is being particularly rude about enjoying their valid salads.

I suppose the main question is, why do you think that most of Security's arsenal dealing damage that borg's cant self-repair would be a good thing? Do you feel Security, or antags that loot the armory, often have trouble with borgs, and that this is the best way to fix it?

**Comment by epochayur** [comment]
>I would argue the real consistency fix would be making borgs immune to lasers altogether.

This is the worst thing I've read all day.

Yes, borgs should take the same damage as everyone else. They don't get a free pass for arbitrary reasons. If you make the decision to play as a silly cone, then you have to live with the ramifications of that. These things aren't the fucking terminator, they're there to carry out the orders of the AI and the crew.

**Comment by XDTM** [comment]
Also consistency-wise fireproof suits are not also immune to lasers

**Comment by PKPenguin321** [comment]
this is fine until borgs get damage slowdown

**Comment by zxaber** [comment]
It's not uncommon for robotics to be staffed by people who do the bare minimum on everything that isn't mechs. Sometimes wire damage is health you will just never get back.

If we're trying to make borgs into hands-less humans, then "borgs should take the same damage as everyone else" should be bundled with "borgs should get to self-heal burn like everyone else". It _is_ only fair, right?

@imsxz 
>To clarify, I wouldn’t have issues with making burn damage just require slapping borgs with a cable.

I had missed this section before, and I apologize for doing so. The gameplay ramifications to borgs wouldn't be as detrimental if this was the case. However, if the purpose of this PR really _is_ to add some realism, consider this; In reality, heat damaged metal is generally cut off and replaced via welding, as are severally bent parts. This is emulated by welding in game. Indeed, it's probably better to think of the current damage system for borgs not in the brute/burn sense (though it uses the same variables), but in the physical/electrical sense. The fluff for wire damage is just rather misleading. And for the record, the same physical/electrical system is also used by mechs.

**Comment by epochayur** [comment]
>If we're trying to make borgs into hands-less humans...

That's not what they are, have been, or ever will be and I seriously have no idea where you're getting that impression from

> then "borgs should take the same damage as everyone else" should be bundled with "borgs should get to self-heal burn like everyone else". It is only fair, right?

No. Borgs are not just humans with slightly different flavor. They are second-class citizens that exist to serve the AI and the crew, and as such should need to rely on them for charging, modifications, etc., in addition to healing.

**Comment by imsxz** [comment]
@zxaber @Arianya it was supposed to be more of a consistency thing than realism, mainly referring to repairing prosthetic limbs(the shit cyborgs are made of), I'd hope it didn't appear I was attempting to mislead this as a realism PR when I personally labeled it balance (mainly a reply to arianya). 

I'd be totally fine with closing this PR for now and opening a new one that revamps exterior burn damage to be simple to repair and also adding in a rarer internal damage (from EMP's, explosions, et cetera. Basically lots of the things that currently cause burn damage to cyborgs could be changed to internal)

**Comment by optimumtact** [comment]
naksu is the only sane silicon player in here, the rest of you are frothers.

Btw you need to fix this properly, not like this, even goof pointed it out

**Comment by imsxz** [comment]
gonna make a PR when i get the motivation that includes this but makes burn damage easier to repair and adds internal cyborg damage so expect that in a few years

**Pull request closed without merging**